partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt

```diff
@@ -759,6 +776,74 @@ internal class PlanTyper(
                 false -> StaticType.ANY
             }
         }
+
+        /**
+         * Resolution and typing of aggregation function calls.
+         *
+         * I've chosen to place this in RexTyper because all arguments will be typed using the same locals.
+         * There's no need to create new RexTyper instances for each argument. There is no reason to limit aggregations
+         * to a single argument (covar, corr, pct, etc.) but in practice we typically only have single <value expression>.
+         *
+         * This method is _very_ similar to scalar function resolution, so it is temping to DRY these two out; but the
+         * separation is cleaner as the typing of NULLS is subtly different.
+         *
+         * SQL-99 6.16 General Rules on <set function specification>
+         *     Let TX be the single-column table that is the result of applying the <value expression>
+         *     to each row of T and eliminating null values <--- all NULL values are eliminated as inputs
+         */
+        public fun resolveAgg(agg: Agg.Unresolved, arguments: List<Rex>): Pair<Rel.Op.Aggregate.Call, StaticType> {
+            var missingArg = false
+            val args = arguments.map {
+                val arg = visitRex(it, null)
+                if (arg.type == MissingType) missingArg = true
+                arg
+            }
+
+            //
+            if (missingArg) {
+                handleAlwaysMissing()
+                return relOpAggregateCall(agg, listOf(rexErr("MISSING"))) to MissingType
+            }
+
+            // Try to match the arguments to functions defined in the catalog
+            return when (val match = env.resolveAgg(agg, args)) {
+                is FnMatch.Ok -> {
+                    // Found a match!
+                    val newAgg = aggResolved(match.signature)
+                    val newArgs = rewriteFnArgs(match.mapping, args)
+                    val returns = newAgg.signature.returns
+
+                    // Determine the nullability of the return type
+                    var isNullable = false // True iff has a NULLABLE arg and is a NULLABLE operator
```

Comment:
Ok. Also, besides count, aggregation returns are _always_ nullable because the input table may be empty.

```
CREATE TABLE numbers (
  n INT NOT NULL
);

SELECT AVG(n) as "avg" FROM numbers;
```

Result is `avg: null`